src/btree/row_srch.c

```diff
@@ -614,14 +620,7 @@ leaf_match:	cbt->compare = 0;
 
 	return (0);
 
-err:	/*
-	 * Release the current page if the search started at the root. If the
-	 * search didn't start at the root we should never have gone looking
-	 * beyond the start page.
-	 */
-	WT_ASSERT(session, leaf == NULL || leaf == current);
-	if (leaf == NULL)
-		WT_TRET(__wt_page_release(session, current, 0));
+err:	WT_TRET(__wt_page_release(session, current, 0));
```

Comment:
@agorrod, I think we still need this code (although the test does change):
```
err:    /*
         * Release the current page if the search started at the root. If the
         * search didn't start at the root we should never have gone looking
         * beyond the start page.
         */
        if (leaf == NULL)
                WT_TRET(__wt_page_release(session, current, 0));
```
otherwise, in the case of searching just a leaf page, we'll release the leaf page on error, which we didn't use to do.